### PR TITLE
Amend way to extract file path on Deployer

### PR DESCRIPTION
**Pull request by boyeon555**
- Amends to replace the first occurrence only of the path which is upper than package.json by using count on the function replace
- Applies it to more than revision 4

### CODE REVIEW

File: iconservice/deploy/icon_score_deployer.py [hunk 1]
```diff
@@ -94,7 +101,7 @@ def _extract_files_gen(data: bytes):
             raise ScoreInstallExtractException(f'Error raising from extract_files_gen: {e}')
 
     @staticmethod
-    def deploy_legacy(path: str, data: bytes):
+    def deploy_legacy(path: str, data: bytes, *args):
```
Comment (leeheonseung) [inline iconservice/deploy/icon_score_deployer.py]: Why did you append this kw_arg?
I think it doesn't need this parameter

Comment (boyeon555) [reply iconservice/deploy/icon_score_deployer.py]: Applied it.

File: iconservice/deploy/icon_score_deployer.py [hunk 2]
```diff
@@ -82,7 +85,11 @@ def _extract_files_gen(data: bytes):
                                 and file_path.find('/.') < 0
                                 and file_path.find(matched_file_path) == 0
                         ):
-                            file_path = file_path.replace(matched_file_path, '')
+                            if revision >= 3:
```
Comment (sink772) [inline iconservice/deploy/icon_score_deployer.py]: Please use the constant, REVISION_3.

File: tests/test_icon_score_deployer.py [hunk 3]
```diff
@@ -132,6 +133,34 @@ def test_deploy_when_score_depth_is_different(self):
             score_path: str = get_score_path(self.score_root_path, address)
             remove_path(score_path)
 
+    def test_deploy_replace_once(self):
+        """
+        Test for replacing the first occurrence only of the path which is upper than package.json
+        by using count on the function `replace`. It is supported on revision_3_or_more.
+        """
+        revision_list = [2, 3, 4]
```
Comment (sink772) [inline tests/test_icon_score_deployer.py]: Please remove the revision 4 since we don't use it yet.

File: tests/test_icon_score_deployer.py [hunk 4]
```diff
@@ -132,6 +133,34 @@ def test_deploy_when_score_depth_is_different(self):
...
+            file_path_list = [name for name, info, parent_dir in zip_file_info_gen]
+            file_path_list.sort()
+            file_path_list_for_revision_2 = ["__init__.py", "score_registry.py"]
+            file_path_list_for_revision_3_or_more = ["score_registry/__init__.py", "score_registry/score_registry.py"]
+            if revision == 2:
+                target_file_path_list_only_for_revision_2 = [file_path for file_path in file_path_list if file_path in file_path_list_for_revision_2]
+                self.assertEqual(file_path_list_for_revision_2, target_file_path_list_only_for_revision_2)
+                target_file_path_list = [file_path for file_path in file_path_list
+                                         if file_path not in file_path_list_for_revision_3_or_more]
+            elif revision > 2:
```
Comment (goldworm-icon) [inline tests/test_icon_score_deployer.py]:
```suggestion
            else:
```

File: tests/test_icon_score_deployer.py [hunk 5]
```diff
@@ -43,12 +40,16 @@ def read_zipfile_as_byte(archive_path: str) -> bytes:
             return byte_data
 
     def test_install(self):
+        self.normal_score_path = os.path.join(DIRECTORY_PATH, 'sample','normal_score.zip')
```
Comment (goldworm-icon) [inline tests/test_icon_score_deployer.py]:
```suggestion
        self.normal_score_path = os.path.join(DIRECTORY_PATH, 'sample', 'normal_score.zip')
```

File: tests/test_icon_score_deployer.py [hunk 6]
```diff
@@ -65,35 +66,36 @@ def test_install(self):
...
-        IconScoreDeployer.deploy(score_deploy_path, self.read_zipfile_as_byte(self.archive_path3))
+        IconScoreDeployer.deploy(score_deploy_path, self.read_zipfile_as_byte(self.innerdir_path))
         self.assertEqual(True, os.path.exists(score_deploy_path))
 
     def test_remove_existing_score(self):
         tx_hash: bytes = create_tx_hash()
         score_deploy_path: str = get_score_deploy_path(self.score_root_path, self.address, tx_hash)
 
-        IconScoreDeployer.deploy(score_deploy_path, self.read_zipfile_as_byte(self.archive_path))
+        self.normal_score_path = os.path.join(DIRECTORY_PATH, 'sample','normal_score.zip')
```
Comment (goldworm-icon) [inline tests/test_icon_score_deployer.py]:
```suggestion
        self.normal_score_path = os.path.join(DIRECTORY_PATH, 'sample', 'normal_score.zip')
```

File: tests/test_icon_score_deployer.py (hunk 3 above)
Comment (boyeon555) [reply tests/test_icon_score_deployer.py]: It has been applied.

File: tests/test_icon_score_deployer.py (hunk 5 above)
Comment (boyeon555) [reply tests/test_icon_score_deployer.py]: It has been applied.

**Pull request merged**